test/IDE/complete_default_arguments_rdar89051832.swift

```diff
@@ -0,0 +1,69 @@
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// RUN: %target-swift-ide-test(mock-sdk: %clang-importer-sdk) -batch-code-completion -source-filename %t/test.swift -filecheck %raw-FileCheck -completion-output-dir %t/out -code-completion-annotate-results -import-objc-header %t/ObjC.h -enable-objc-interop %t/Lib.swift 
+// REQUIRES: objc_interop
+
+
+//--- ObjC.h
+@import Foundation;
+
+typedef NS_OPTIONS(NSInteger, MyOptions) {
+  MyOptOne = 1 << 0,
+  MyOptTwo = 1 << 1,
+};
+
+@interface MyObj : NSObject
+// 'opt' should not be defaulted.
+// FIXME: Currently this is considered defaulted because the base name is 'store'.
+- (void)storeOptions:(MyOptions)opts;
```

Comment:
Seems like it wouldn't be too hard to do this. The default isn't added based on the function name at all - the it's added by checking the name of the type. Ie. if it's a enum type with Options kind and the name contains `options` (split by camel case and ignoring the first case) then we add a default.

Reply:
Checking (1) it's the first parameter (2) it's 'xxxOptions' enum type, is not not enough. We need to differentiate between:
```objc
-(void)performWithOptions: (MyOptions) opts;
-(void)storeOptions: (MyOptions) opts;
```